cms-publish-worker-webapp/src/main/java/se/simonsoft/cms/publish/worker/PublishJobService.java

```diff
@@ -141,24 +193,58 @@ private String formatParam(String param, String pathName) {
 		}
 		return result;
 	}
-	
+
 	public boolean isCompleted(PublishTicket ticket) throws PublishException {
 		logger.debug("Checking if job with ticket: {} is done", ticket.toString());
 		PublishRequestDefault request = new PublishRequestDefault();
 		request.addConfig("host", this.publishHost);
 		request.addConfig("path", this.publishPath);
 		return pe.isCompleted(ticket, request);
 	}
-	
+
 	private String writeToTmpFile(PublishJobOptions jobOptions, PublishTicket ticket, PublishRequestDefault request) throws IOException, PublishException {
-		
+
 		File temp = File.createTempFile(jobOptions.getPathname(), ".tmp");
 		FileOutputStream fout = new FileOutputStream(temp);
 		pe.getResultStream(ticket, request, fout);
 		logger.debug("Temporary file placed at: {}", temp.getAbsolutePath());
 		fout.flush();
 		fout.close();
-		
+
 		return temp.getAbsolutePath();
 	}
+
+	private String retrieveSource(PublishJobOptions jobOptions) {
+		String jobName = jobOptions.getStorage().getPathconfigname();
+		jobName += jobOptions.getStorage().getPathdir();
+		jobName += "/" + jobOptions.getStorage().getPathnamebase();
+		CmsExportPrefix cmsPrefix = new CmsExportPrefix(jobOptions.getStorage().getPathversion());
+		String cloudId = jobOptions.getStorage().getPathcloudid();
+		String bucketName = jobOptions.getStorage().getParams().get("s3bucket");
+		CmsExportProviderAwsSingle exportProvider = new CmsExportProviderAwsSingle(cmsPrefix, cloudId, bucketName, region, credentials);
+		CmsImportJob downloadJob = new CmsImportJobSingle(null, jobName, "preprocess.zip");
```

Comment:
Is it possible to use?
PublishExportJobFactory.getImportJobSingle(jobOptions.getStorage(), "preprocess.zip");

Reply:
In the POM, upgrade cms-publish-config to 0.5.0-SNAPSHOT. That will expose the factory.

Reply:
Yep, done.